src/charm.py

```diff
@@ -158,17 +154,14 @@ def _generate_root_certificate(self) -> None:
         """
         if not self._config_ca_common_name:
             raise ValueError("CA common name should not be empty")
-        private_key_password = generate_password()
-        private_key = generate_private_key(password=private_key_password.encode())
+        private_key = generate_private_key()
         ca_certificate = generate_ca(
             private_key=private_key,
             subject=self._config_ca_common_name,
```

Comment:
I think it's worth adding the validity here and not rely on the default value.
It should come from `_config_root_ca_certificate_validity`

Reply:
I agree but this is unrelated to this change right? I would prefer to do it in a separate PR

Reply:
Issue created. #213